service/api_patch.go

```diff
@@ -109,8 +111,46 @@ func (as *APIServer) submitPatch(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	patchID := mgobson.NewObjectId()
+	author := dbUser.Id
+	grip.Debug(message.Fields{
+		"message":       "testing user",
+		"github_author": data.GithubAuthor,
+	})
+	if data.GithubAuthor != "" {
+		opts := gimlet.PermissionOpts{
+			Resource:      pref.Id,
+			ResourceType:  evergreen.ProjectResourceType,
+			Permission:    evergreen.PermissionPatches,
+			RequiredLevel: evergreen.PatchSubmitAdmin.Value,
+		}
+		if !dbUser.HasPermission(opts) {
+			as.LoggedError(w, r, http.StatusUnauthorized, errors.New("user is not authorized to patch on behalf of other users"))
+			return
+		}
+		specifiedUser, err := user.FindByGithubName(data.GithubAuthor)
+		if err != nil {
+			as.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "error looking for specified author"))
```

Comment:
In this case, shouldn't we just use an anonymous author instead of erroring?

Reply:
I think it's correct to error for database issues (this is rare anyway).